### ISSUE TITLE
Configure max_trace_back= 0 returns full exception stack

### ISSUE THREAD

**Issue by chanchiem** [aws/aws-xray-sdk-python#120]
Currently, if the recorder is configured to have a max_trace_back of 0, the full exception trace is returned on its subsegment. If the max_trace_back is 1, only one line of the exception trace is returned. 

This is an unexpected behavior and should instead return an empty trace stack if max_trace_back is configured to be 0.

**Comment by chanchiem** [comment]
Found the root cause of the issue:
The recorder incorrectly checks to see if the max_trace_back is not none through the following call:
```
if max_trace_back:
    self.max_trace_back = max_trace_back
```

Therefore, if max_trace_back = 0, the above would be executed as if False/None. 

To fix, check if it's not none and if it's greater than or equal to 0.

**Comment by chanchiem** [comment]
A deeper problem has been found. When the utility class Throwable is constructed in the entity:
https://github.com/aws/aws-xray-sdk-python/blob/master/aws_xray_sdk/core/models/entity.py#L226

it doesn't add "stack" as an attribute to the throwable object. This is problematic during serialization and causes a "'Throwable' object has no attribute 'stack'" exception in the entity.py serialization: 
https://github.com/aws/aws-xray-sdk-python/blob/master/aws_xray_sdk/core/models/entity.py#L238

It looks like the solution is to not set the throwable stack attribute to None using the pythonic check:
https://github.com/aws/aws-xray-sdk-python/blob/master/aws_xray_sdk/core/models/throwable.py#L51

But, rather, use explicit type checking to ensure that empty lists don't return None. More will be updated as this is more investigated and tested.

**Comment by chanchiem** [comment]
Fixed via PR #123
